flink/build.gradle

```diff
@@ -28,6 +28,7 @@ project(':iceberg-flink') {
     implementation project(':iceberg-parquet')
     implementation project(':iceberg-hive-metastore')
 
+    compileOnly "org.apache.flink:flink-connector-base"
```

Comment:
It's strange the the build for flink 1.12 & flink 1.13 has been passed, because I don't see the same dependency are added to flink 1.12 build.gradle and 1.13 build.gradle.  Maybe I need to check the 1.12's build.gradle again.

Reply:
@openinx Maybe follow-up on the other [comment discussion](https://github.com/apache/iceberg/pull/3354/files#r740053509) here.

With the SplitEnumerator API change, looks like I need to put FLIP-27 source in the `v1.13` folder. What should we do with future versions (like 1.14)? do we copy the FLIP-27 source code `v1.13` to `v1.14` folder?

Reply:
@openinx, the tests run against the iceberg-flink module. They aren't present in the 1.12 or 1.13 modules. If you want them to be run for those modules, you'd need to add the source folder like you do for `src/main/java`. If you choose to do that, let's also remove CI for the common module since we don't need to run the tests outside of 1.12 and 1.13 if they are run in those modules.

Reply:
@stevenzwu, I think that copying the parts that change is reasonable. And once we remove support for 1.12, you can move the files back into the common module.

Reply:
yeah. that is my plan too. Once 1.12 support is removed, we should be able to move files back to the common module. We just need to be diligent with these efforts.